### ISSUE TITLE
CldVideoPlayer fullscreen object-cover issue

### ISSUE THREAD

**Issue by apetta** [cloudinary-community/next-cloudinary#433]
I'm having a challenge getting a full-screen video with object-cover to work with `CldVideoPlayer` in nextjs 14. I think the styles from `next-cloudinary/dist/cld-video-player.css` may be causing an issue here. A regular `video` element works as expected.

This is on `"next-cloudinary": "~5.20.0"`

**Comment by colbyfayock** [comment]
hey @apetta can you share the code you're trying to use and an example of what you're trying to achieve?

**Comment by apetta** [comment]
@colbyfayock Sure thing! Here's a [repo with the issue ](https://github.com/apetta/cloudinary-player-bug)

You can see the issue on a desktop viewport.

Desired layout:
![Screenshot 2024-02-20 at 11 51 17](https://github.com/cloudinary-community/next-cloudinary/assets/33671231/8348a867-9878-417d-a4c5-b54df43b46bc)

With next-cloudinary:
![Screenshot 2024-02-20 at 11 51 32](https://github.com/cloudinary-community/next-cloudinary/assets/33671231/eda780fe-86ea-41fb-a905-1621c0e69311)

Thank you!

**Comment by colbyfayock** [comment]
thanks for the example, that was really helpful

i think this is currently simply not supported natively. I'm currently wrapping the player with an aspect ratio'ed container in order to provide better responsiveness and id imagine that's certainly preventing this ability

looking into the issue itself though, the video player uses videojs under the hood, and found some potential solutions here: https://github.com/videojs/video.js/issues/3431

id be happy to rethink that and add this capability, but currently working on wrapping up a major release, so would need to be once im through with that: https://github.com/cloudinary-community/next-cloudinary/pull/407

that said, i think there are two options in the meantime:
- you may be able to figure out a hacky solution using absolute positioning or a little javascript to make a video fill its container and hide the overflow

along the lines of:
```
<div that fills the container and hides overflow>
  <CldVideoPlayer
</div>
```

- or you could use getCldVideoUrl to generate the URL and use the native video tag, though im unsure if there are specific features with the Video Player you're hoping to maintain

https://next.cloudinary.dev/getcldvideourl/basic-usage

**Comment by colbyfayock** [comment]
yeah im thinking that because im not exposing anything that allows you to modify the video element that won't be possible

yes please, can you create a new ticket for that one?

**Comment by Zakisb** [comment]
hey @apetta  did you find a solution? i'm having the same issue right now !

**Comment by apetta** [comment]
> hey @apetta  did you find a solution? i'm having the same issue right now ! 

Hey, I think I just generated the URL and used a custom `video` element in the end

**Comment by ARYPROGRAMMER** [comment]
greetins @jlooper-cloudinary , can i get this assigned. wouldnt take much time i think given the previous efforts

**Comment by ArjunCodess** [comment]
hey! can i get this assigned to me?

**Comment by ArjunCodess** [comment]
please assign this to me. im almost done with the code part.

**Comment by ArjunCodess** [comment]
just did a PR for this!
